plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsFrontend.php

```diff
@@ -186,42 +186,60 @@ public function edit_account_form_fields() {
 	 * @param integer $user_id User ID.
 	 */
 	public function save_account_form_fields( $user_id ) {
-		// phpcs:disable WordPress.Security.NonceVerification.Missing
-		$customer          = new WC_Customer( $user_id );
-		$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
-		$field_values      = array();
 
-		foreach ( array_keys( $additional_fields ) as $key ) {
-			$post_key = CheckoutFields::get_group_key( 'other' ) . $key;
-			if ( ! isset( $_POST[ $post_key ] ) ) {
-				continue;
-			}
+		try {
+			$customer = new WC_Customer( $user_id );
 
-			$field_value = $this->checkout_fields_controller->sanitize_field( $key, wc_clean( wp_unslash( $_POST[ $post_key ] ) ) );
-			$validation  = $this->checkout_fields_controller->validate_field( $key, $field_value );
+			$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
 
-			if ( is_wp_error( $validation ) && $validation->has_errors() ) {
-				wc_add_notice( $validation->get_error_message(), 'error' );
-				continue;
+			$field_values = array();
+
+			foreach ( $additional_fields as $key => $field_data ) {
+				// We can't skip, field might be required.
+				$field_value = trim( wc_clean( wp_unslash( $_POST[ $key ] ?? '' ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+
+				if ( 0 === strlen( $field_value ) && ! ( $field_data['required'] ?? false ) ) {
+					continue; // Skip, otherwise validation might prevent saving empty optional fields.
```

Comment:
I think this would be the solution for saving empty optional fields.
```suggestion
					$field_values[ $key ] = '';
					continue; // Skip, otherwise validation might prevent saving empty optional fields.
```

Reply:
Thanks @opr, indeed this was missing. I also updated tests to check for saved values wherever I saw fit, so this would be caught!